File: app/src/main/java/com.example.petridelivery.wrappers/base/abs/File.java
package com.example.petridelivery.wrappers.base.abs;

public class File {
    private String fileName;

    public File(String fileName) {
        setFileName(fileName);
    }

    public void setFileName(String fileName) {
        if(fileName.contains("/")){
            throw new IllegalArgumentException("file cannot have forward slash");
        }
        this.fileName = fileName;
    }

    @Override
    public String toString() {
        return fileName;
    }
}

File: app/src/main/java/com.example.petridelivery.wrappers/base/abs/File.kt
package com.example.petridelivery.wrappers.base.abs

class File(private var fileName: String) {

    fun setFileName(fileName: String) {
        require(!fileName.contains("/")) { "file cannot have forward slash" }
        this.fileName = fileName
    }

    override fun toString(): String {
        return fileName
    }

    init {
        setFileName(fileName)
    }
}